src/api/app/models/user.rb

```diff
@@ -244,9 +244,10 @@ def self.admin_session?
     current && current.is_admin?
   end
 
-  # set the user as current session user (should be real user)
+  # set the user as current session user (should be real user or nil)
   def self.session=(user)
     Thread.current[:user] = user
+    Feature.use_beta_features(user.try(:in_beta) || false)
```

Comment:
We use ```User.session``` everywhere, specially models, model specs and API. I don't think this WebUI specific setting should be necessary for them..

Reply:
what makes you think features are webui specific? It's just a question of time til (staging) API enables some features only for some users.

But even if not: The use_beta_features function only resets some variables. And User.session= is the right place IMO